root/usr/bin/discovery-register

```diff
@@ -63,20 +63,23 @@ facts = Hash[Facter.to_hash.select {|k,v| k =~ /address|hardware|manufacturer|pr
 facts.keys.sort.each {|k| log_msg " #{k}: #{facts[k]}"}
 
 # check every 15 minutes but only upload on changes
-i = cmdline("fdi.cachefacts", 0).to_i rescue 0
 upload_sleep = cmdline("fdi.uploadsleep", 60 * 15).to_i rescue (60 * 15)
+extra_initial_uploads = cmdline("fdi.cachefacts", 0).to_i rescue 0
 while true do
   uninteresting_facts=/kernel|operatingsystem|osfamily|ruby|path|time|swap|free|filesystem|version|selinux/i
+  # force facter to do new system scan (we default to 15 minute intervals which is safe)
+  Facter.clear
```

Comment:
This is essentially the only change, the rest is just renamed local variable, shuffling with order of lines for better readability and an extra debug message.